src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTestCommon/EndpointInfoSourceCallback.cs

```diff
@@ -89,6 +100,19 @@ public Task OnRemovedEndpointInfoAsync(IEndpointInfo info, CancellationToken tok
                 token);
         }
 
+        private static Task ApplyStartupHookAsync(DiagnosticsClient client, string path, CancellationToken token)
+        {
+            // DiagnosticsClient.ApplyStartupHookAsync currently is not public
+            MethodBase applyStartupHookAsync =
+                typeof(DiagnosticsClient).GetMethod(
```

Comment:
Will this become public in the future?

Reply:
I hope so. Many of the APIs that we've added are marked internal pending API review. Maybe we can make an effort to make them all public in the .NET 8 timeframe.